### ISSUE TITLE
The api_id/api_hash combination is invalid

### ISSUE THREAD

**Issue by tomillo** [LonamiWebs/Telethon#836]
It turns out that my script worked perfectly until I tried using it on another server, and from that moment on it stopped working and the error appears **api_id / api_hash combination is invalid**. I have checked api_id and hash_id many times, but they haven't changed.. so I would appreciate some help. Thanks.

```python
Traceback (most recent call last):
  File "mytelegram.py", line 41, in <module>
    tclient.send_code_request(phone)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/telethon/telegram_client.py", line 241, in send_code_request
    result = self(SendCodeRequest(phone, self.api_id, self.api_hash))
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/telethon/telegram_bare_client.py", line 486, in __call__
    result = self._invoke(call_receive, request, ordered=ordered)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/telethon/telegram_bare_client.py", line 583, in _invoke
    raise next(x.rpc_error for x in requests if x.rpc_error)
telethon.errors.rpc_error_list.ApiIdInvalidError: The api_id/api_hash combination is invalid
```

**Comment by Lonami** [comment]
See #46, we already had issues with this. Specifically [this comment](https://github.com/LonamiWebs/Telethon/issues/46#issuecomment-293614461):
> As far as i know this problem has something to do with the  length of new API keys. All API keys that are 5 digits or less - works  perfectly. But new ones has 6 digits. And they fail to authorize.

Does your `api_id` have over 6 digits? Maybe, we had issues when they started having 6 digits, perhaps we have them again with them being 7. It wouldn't surprise me. My `api_id/api_hash` (with 5 digits) works just fine.

**Comment by tomillo** [comment]
Thanks @Lonami for your reply. My `api_id`have 6 digits.

**Comment by tomillo** [comment]
Still with the same error. Any help?

**Comment by tomillo** [comment]
There’s a way to renew API ID/hash with the same account?

**Comment by Lonami** [comment]
No.

**Comment by zsamora** [comment]
check for spaces, mine was not working because of a blank space in the beginning

**Comment by alvaropc** [comment]
Same problem here.
Telegram keeps giving IDs with more than 5 digits, why this cannot be solved?

**Comment by Lonami** [comment]
> Telegram keeps giving IDs with more than 5 digits, why this cannot be solved?

Because there are more than 99_999 developers, there can't be new IDs with less digits. If you're talking about the bug itself, there is not much the library can do. If Telegram rejects an API ID, the best you can do is either wait, email support, or try to get a new API ID in another account.

**Comment by kylethedeveloper** [comment]
This is so ridiculous.

**Comment by Lonami** [comment]
Yes, but there's nothing Telethon can do about it. Waiting for a while after creating the app may make the problem get solved on its own. Not sure, I only have one phone number to make apps with, and I did this a really long time ago.

**Comment by kylethedeveloper** [comment]
I actually figured it out. It was my fault I guess.

I created a bot account from BotFather and I was using my bot's API token for all the variables; `api_id`, `api_hash` and `bot_token`.
I did not realize that I had to use [**my**](https://my.telegram.org/apps) `api_id` and `api_hash` for those variables. So to summarize:
```python
api_id = <youraccount_api_id>
api_hash = <youraccount_api_hash>
bot_token = '123456789:apitoken_ofyourbot_thatyougetfrom_botfather'
```

I was actually going back and forth through the documentation and realized that I was wrong after reading [TelegramClient](https://docs.telethon.dev/en/latest/modules/client.html) page. In my opinion it should be mentioned on [here](https://docs.telethon.dev/en/latest/basic/signing-in.html#signing-in-as-a-bot-account) too.

> BTW both my own api_id and bot's api_id are longer than 5 chars.

**Comment by Lonami** [comment]
The "here" you mention already says:

> *(at the top)* **Note**: This API ID and hash is the one used by your *application*, not your phone number. You can use this API ID and hash with any phone number or even for bot accounts.
>
> **Signing In as a Bot Account** […] You will still need an API ID and hash.

API ID, API hash, and bot token, all are different terms. But if you can improve the documentation, feel free to send a PR.

**Comment by Dhinendran** [comment]
@kylethedeveloper I have all the scenarios both key and mobile number  throwing 

`telethon.errors.rpcerrorlist.ApiIdInvalidError: The api_id/api_hash combination is invalid (caused by SendCodeRequest)`

any on help me out. my api_id is 7 digit

**Comment by shtff** [comment]
> @kylethedeveloper I have all the scenarios both key and mobile number throwing
> 
> `telethon.errors.rpcerrorlist.ApiIdInvalidError: The api_id/api_hash combination is invalid (caused by SendCodeRequest)`
> 
> any on help me out. my api_id is 7 digit

same problem

**Comment by slkass** [comment]
> @kylethedeveloper I have all the scenarios both key and mobile number throwing
> 
> `telethon.errors.rpcerrorlist.ApiIdInvalidError: The api_id/api_hash combination is invalid (caused by SendCodeRequest)`
> 
> any on help me out. my api_id is 7 digit

same

**Comment by sdfg2** [comment]
I have the same problem with a 7 digit id.  Is it a case of waiting like it was with #46 ?

**Comment by sgerodes** [comment]
I have a 9 digit long API_ID and it is also not working.

**Comment by lion2131** [comment]
any updates on the topic?

**Comment by hayuntal** [comment]
Any updates about this error?

**Comment by webyneter** [comment]
Any progress on this one recently?

**Comment by Jammydodgy** [comment]
Set the API details in the environment, this solved the problem for me

import os
os.environ["API_ID"] = ""
os.environ["API_HASH"] = ""
os.environ["PHONE_NUMBER"] = ""


Access in the code as below:

API_ID = int(os.getenv("API_ID"))  # Convert to int
API_HASH = os.getenv("API_HASH")
PHONE_NUMBER = os.getenv("PHONE_NUMBER")

**Comment by FantasySwords** [comment]
using production address saved me.

` 
client.session.set_dc(2, '149.154.175.100', 80)
`

code from ios telegram swift

`
 seedAddressList = [
                    1: ["149.154.175.50", "2001:b28:f23d:f001::a"],
                    2: ["149.154.167.50", "95.161.76.100", "2001:67c:4e8:f002::a"],
                    3: ["149.154.175.100", "2001:b28:f23d:f003::a"],
                    4: ["149.154.167.91", "2001:67c:4e8:f004::a"],
                    5: ["149.154.171.5", "2001:b28:f23f:f005::a"]
                ]
`

**Comment by ZherebcovSergey** [comment]
I faced the same issue: using my code locally and even with a VPN did not work — always received the "api_id/api_hash combination is invalid" error. However, once I deployed exactly the same script on an AWS production server, it worked without any changes. This makes me think Telegram might have a whitelist for certain IP ranges (like production cloud server IPs) or, conversely, might restrict or block residential, VPN, or other suspicious endpoints. So if the connection is coming from a home network, VPN, or random IP, Telegram could be silently rejecting the authentication request. On the cloud (AWS), everything worked smoothly.
